runtime/jcl/common/thread.cpp

```diff
@@ -385,4 +392,74 @@ Java_java_lang_Thread_startImpl(JNIEnv *env, jobject rcv, jlong millis, jint nan
 	vmFuncs->internalExitVMToJNI(currentThread);
 }
 
+#if JAVA_SPEC_VERSION >= 19
+/* private static native void registerNatives(); */
+void JNICALL
+Java_java_lang_Thread_registerNatives(JNIEnv *env, jclass clazz)
+{
+	return;
+}
+
+/* static native Thread currentCarrierThread(); */
+jobject JNICALL
+Java_java_lang_Thread_currentCarrierThread(JNIEnv *env, jclass clazz)
+{
+	J9VMThread *currentThread = (J9VMThread*)env;
+	J9InternalVMFunctions *vmFuncs = currentThread->javaVM->internalVMFunctions;
+
+	vmFuncs->internalEnterVMFromJNI(currentThread);
+	jobject result = vmFuncs->j9jni_createLocalRef(env, currentThread->threadObject);
```

Comment:
this may not actuall work  if its called by a virtual thread. For the purposes of this PR you can just assert here. No code should be calling this until the preview flag is turned on. So we will only need it to run loom specific tests

Reply:
This is actually used during Thread initialization which requires a basic impl to operate

Reply:
Thanks, I didnt realize that code was using it